src/Uno.UI/UI/Xaml/Media/Animation/Timeline.cs

```diff
@@ -164,21 +164,22 @@ internal BindingPath PropertyInfo
 		{
 			get
 			{
-				if (_propertyInfo == null)
-				{
-					InitTarget();
-					var target = Target ?? GetTargetFromName();
+				// Don't use the cached _propertyInfo if TargetProperty or Target has the changed.
+				var path = Storyboard.GetTargetProperty(this);
+				InitTarget();
+				var target = Target ?? GetTargetFromName();
 
+				if (_propertyInfo == null || _propertyInfo.Path != path)
+				{
```

Comment:
It's best to remove references to an existing `BindingPath` (particularly w.r.t INPC and DP changed registrations)

`_propertyInfo?.DataContext = null;`

Reply:
You mentioned datacontext changes in the last commit message, but I don't see it.

Reply:
@jeromelaban The last commit has `_propertyInfo?.DataContext = null;` line that you suggested. Did I misunderstand your suggestion?

Reply:
Ah yes, looks like github diffs was behaving oddly.